tests/python/04_swizzle/main.py

```diff
@@ -45,108 +44,50 @@ def runKernel(opt):
         if remainder == 3:
             reference.append(idx-2)
 
-    if xclSyncBO(opt.handle, boHandle, xclBOSyncDirection.XCL_BO_SYNC_BO_TO_DEVICE, DATA_SIZE, 0):
-        return 1
 
-    p = xclBOProperties()
-    bodevAddr = p.paddr if not (xclGetBOProperties(opt.handle, boHandle, p)) else -1
+    obj.sync(pyxrt.xclBOSyncDirection.XCL_BO_SYNC_BO_TO_DEVICE, size, 0)
 
-    if bodevAddr is -1:
-        return 1
+    # Create a run object without starting kernel
+    run = pyxrt.run(swizzle);
 
-    # Allocate the exec_bo
-    execHandle = xclAllocBO(opt.handle, DATA_SIZE, 0, (1 << 31))
-    execData = xclMapBO(opt.handle, execHandle, True)  # returns mmap()
-
-    print("Construct the exec command to run the kernel on FPGA")
-
-    xclOpenContext(opt.handle, opt.xuuid, 0, True)
-    # construct the exec buffer cmd to start the kernel
-    start_cmd = ert_start_kernel_cmd.from_buffer(execData.contents)
-    rsz = (XVECTORSWIZZLE_CONTROL_ADDR_A_DATA / 4 + 1) + 1  # regmap array size
-    ctypes.memset(execData.contents, 0, ctypes.sizeof(ert_start_kernel_cmd) + rsz*4)
-    start_cmd.m_uert.m_start_cmd_struct.state = 1  # ERT_CMD_STATE_NEW
-    start_cmd.m_uert.m_start_cmd_struct.opcode = 0  # ERT_START_CU
-    start_cmd.m_uert.m_start_cmd_struct.count = 1 + rsz
-    start_cmd.cu_mask = 0x1
-
-    # Prepare kernel reg map
-    new_data = (ctypes.c_uint32 * rsz).from_buffer(execData.contents, 8)
-    new_data[XVECTORSWIZZLE_CONTROL_ADDR_AP_CTRL] = 0x0 # ap_start
-    new_data[XVECTORSWIZZLE_CONTROL_ADDR_GROUP_ID_X_DATA/4] = 0x0 # group id
-    new_data[XVECTORSWIZZLE_CONTROL_ADDR_A_DATA / 4] = bodevAddr
-    new_data[XVECTORSWIZZLE_CONTROL_ADDR_A_DATA / 4 + 1] = (bodevAddr >> 32) & 0xFFFFFFFF # s1 buffer
-
-    global_dim = [DATA_SIZE / 4, 0]; # int4 vector count global range
+    global_dim = [size // 4, 0]; # int4 vector count global range
     local_dim = [16, 0]; # int4 vector count global range
-    groupSize = global_dim[0] / local_dim[0];
-
-    if opt.verbose == 1:
-        print("Global range: %d " % global_dim[0])
-        print("Group size  : %d " % local_dim[0])
-        print("Starting kernel...\n")
-
+    group_size = global_dim[0] // local_dim[0];
 
-    for id in range(groupSize):
-        if opt.verbose == 1:
-            print("group id = %d" % id)
+    # Run swizzle with 16 (local[0]) elements at a time
+    # Each element is an int4 (sizeof(int) * 4 bytes)
+    # Create sub buffer to offset kernel argument in parent buffer
+    local_size_bytes = local_dim[0] * ctypes.sizeof(ctypes.c_int) * 4;
+    for id in range(group_size):
+        subobj = pyxrt.bo(obj, local_size_bytes, local_size_bytes * id)
+        run.set_arg(0, subobj)
+        run.start()
+        state = run.state()
```

Comment:
It would be nice to simplify your API like
```suggestion
        state = run.set_arg(0, subobj).start().state()
```

Reply:
I wanted to make it easy to step through the code in pdb3 debugger, hence separate statements